src/main/kotlin/de/bund/digitalservice/useid/events/EventService.kt

```diff
@@ -4,22 +4,28 @@ import mu.KotlinLogging
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
 import org.springframework.http.codec.ServerSentEvent
 import org.springframework.stereotype.Service
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter
 import java.util.UUID
-import java.util.function.Consumer
+import java.util.concurrent.Executors
 
 @Service
 @ConditionalOnProperty(name = ["features.desktop-solution-enabled"], havingValue = "true")
 class EventService {
     private val log = KotlinLogging.logger {}
 
-    private val consumers: MutableMap<UUID, Consumer<ServerSentEvent<Any>>> = HashMap()
+    private val consumers: MutableMap<UUID, SseEmitter> = HashMap()
+
+    private val nonBlockingService = Executors.newCachedThreadPool() // TODO: set size of thread pool if necessary
```

Comment:
Open TODO. We should either create a ticket or do it before prod deployment. Wdyt?

Reply:
This is not a core function of our application, right? (is it a desktop prototype?) if yes then maybe it's not a prod deployment blocker (since it's behind the feature flag as well)

Reply:
You are totally right. Let's leave it like that for now.